engine/core.go

```diff
@@ -34,11 +36,15 @@ func (d BaseDeploymentConfig) Version() string {
 	return d.Version_
 }
 
+func (d BaseDeploymentConfig) PreDeployCommand() []string {
+	return d.PreDeployCommand_
+}
+
 func (d BaseDeploymentConfig) PostDeployCommand() []string {
 	return d.PostDeployCommand_
 }
 
-func (d *BaseDeploymentConfig) SetVersion(version string) {
```

Comment:
Did you try this change with the `ploy update` command? I know that Golang best practices state not to mix regular receivers with pointer receivers, but in my memory this was the only thing that worked to actually get the `update` command to work

Reply:
I checked this out. Now the `ploy update` command doesn't work anymore 😅 I suggest we ignore the language purists